tests/x11/nis_server.pm

```diff
@@ -126,15 +125,18 @@ sub run {
     nis_server_configuration();
     wait_serial("$module_name-0", 360) || die "'yast2 nis server' didn't finish";
     assert_screen 'yast2_closed_xterm_visible';
-    # NIS Server is configured and running, configuration continues on client side
-    mutex_create('nis_ready');
     $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => 'nfs_server');
     nfs_server_configuration();
     wait_serial("$module_name-0", 360) || die "'yast2 nfs server' didn't finish";
     assert_screen 'yast2_closed_xterm_visible', 200;
-    # NFS Server is configured and running, configuration continues on client side
-    mutex_create('nfs_ready');
-    wait_for_children;
+    # In order for the hostname to get the set value via yast2 nis_server, a restart is needed.
+    # Otherwise "make" command won't work as in Makefile, there is a variable that gets it's value from "domainname" command
+    systemctl 'restart network';
+    # NIS and NFS Server is configured and running, configuration continues on client side
+    mutex_create('server_ready');
```

Comment:
I would suggest to use more unique name for the lock. This one is use in other test too. I doubt that those two will get mixed ever, but we can eliminate this risk.

Reply:
Done.